collect_app/src/main/java/org/odk/collect/android/widgets/items/SelectOneFromMapWidget.kt

```diff
@@ -22,10 +23,18 @@ import org.odk.collect.androidshared.ui.DialogFragmentUtils
 import org.odk.collect.permissions.PermissionListener
 
 @SuppressLint("ViewConstructor")
-class SelectOneFromMapWidget(context: Context, questionDetails: QuestionDetails) :
-    QuestionWidget(context, questionDetails), WidgetDataReceiver {
+class SelectOneFromMapWidget(
+    context: Context,
+    questionDetails: QuestionDetails,
+    private val autoAdvance: Boolean
+) : QuestionWidget(context, questionDetails), WidgetDataReceiver {
+
+    lateinit var autoAdvanceListener: AdvanceToNextListener
```

Comment:
Could we just pass in or set the `autoAdvanceListener` rather than piggybacking it on `context`? I know other widgets use this trick, but I think I'd prefer if this worked more like the widget change listener so there's only one way (and an explicit) way of setting it.

Reply:
Done.